apps/console/src/app/app.tsx

```diff
@@ -18,32 +20,32 @@ import {
 import { useAuth } from '@console/shared/auth'
 import { useAuthInterceptor, useDocumentTitle } from '@console/shared/utils'
 import { LoadingScreen } from '@console/shared/ui'
+import { AppDispatch } from '@console/store/data'
 import { OverviewPage } from '@console/pages/overview/feature'
 import { SettingsPage } from '@console/pages/settings/feature'
-import { OnboardingPage } from '@console/pages/onboarding/feature'
+import { PageOnboarding } from '@console/pages/onboarding'
 import { InfraLogsPage } from '@console/pages/logs/infra/feature'
 import { PageLogin } from '@console/pages/login'
-
-import { useProjects } from '@console/domains/projects'
+import { fetchProjects } from '@console/domains/projects'
 import { environment } from '../environments/environment'
 import { Layout } from '@console/pages/layout'
 import { PageServices } from '@console/pages/services'
 import { PageApplication } from '@console/pages/application'
 import { PageEnvironments } from '@console/pages/environments'
 
 function RedirectOverview() {
-  const { organizationId } = useParams()
-  const { getProjects } = useProjects()
+  const { organizationId = '' } = useParams()
+  const dispatch = useDispatch<AppDispatch>()
 
   useEffect(() => {
-    async function fetchProjects() {
-      const projects: any = organizationId && (await getProjects(organizationId))
-      if (projects?.payload.length > 0) {
-        window.location.href = OVERVIEW_URL(organizationId, projects.payload[0].id)
+    async function fetchCurrentProjects() {
+      const projects: Project[] = await dispatch(fetchProjects({ organizationId })).unwrap()
```

Comment:
Interesting this unwrap thing. It works okay here so no need to change it but in the rest of the project, I prefer we write better one dispatch in a useEffect, and one useSelector to read the projects from the store and then you can do a useEffect that make an action if the projects are changing

Reply:
Yes I am agree with this approach